lib/component.js

```diff
@@ -396,6 +412,34 @@ class Component extends WebComponent {
     return state;
   }
 
+  _updateAttrs(attr) {
+    const attrsSchema = this.constructor.attrsSchema;
+    if (attrsSchema.hasOwnProperty(attr)) {
+      const propName = attrsSchema[attr].propName || this._toCamelCase(attr);
+      const propType = attrsSchema[attr].type || `string`;
+      let propValue = null;
+
+      if (propType === `string`) {
```

Comment:
I don't feel super strongly about this, but do you have a reason for preferring string type names to the [Vue-style types](https://vuejs.org/v2/guide/components-props.html#Prop-Types) @zakj and I have suggested in the past where this would be `if (propType === String)` etc?

Reply:
I personally like strings because the entire attrSchema return can be jsoni-fied. This means I could create a component called `<attr-schema-doc component-name='hello world' attrs-schema="{schema json here}">`.